File: PomodoroTimer.cs
using System;
using System.Windows.Forms;
using System.Drawing;

namespace PomodoroTimer
{
    public class PomodoroTimerForm : Form
    {
        private Timer timer;
        private Label timerLabel;
        private Button startButton;
        private Button resetButton;
        private Label modeLabel;
        private int workMinutes = 25;
        private int breakMinutes = 5;
        private bool isWorking = true;
        private bool isRunning = false;

        public PomodoroTimerForm()
        {
            // フォームの設定
            this.Text = "ポモドーロタイマー";
            this.Size = new Size(300, 200);
            this.StartPosition = FormStartPosition.CenterScreen;

            // タイマーラベル
            timerLabel = new Label
            {
                Text = "25:00",
                Font = new Font("Arial", 48),
                TextAlign = ContentAlignment.MiddleCenter,
                Size = new Size(200, 80),
                Location = new Point(50, 20)
            };
            this.Controls.Add(timerLabel);

            // スタート/ストップボタン
            startButton = new Button
            {
                Text = "開始",
                Size = new Size(100, 30),
                Location = new Point(50, 120)
            };
            startButton.Click += StartButton_Click;
            this.Controls.Add(startButton);

            // リセットボタン
            resetButton = new Button
            {
                Text = "リセット",
                Size = new Size(100, 30),
                Location = new Point(150, 120)
            };
            resetButton.Click += ResetButton_Click;
            this.Controls.Add(resetButton);

            // モードラベル
            modeLabel = new Label
            {
                Text = "作業時間",
                Font = new Font("Arial", 12),
                TextAlign = ContentAlignment.MiddleCenter,
                Size = new Size(200, 30),
                Location = new Point(50, 100)
            };
            this.Controls.Add(modeLabel);

            // タイマーオブジェクト
            timer = new Timer
            {
                Interval = 1000  // 1秒ごとに更新
            };
            timer.Tick += Timer_Tick;
        }

        private void Timer_Tick(object sender, EventArgs e)
        {
            if (isWorking)
            {
                workMinutes--;
                if (workMinutes < 0)
                {
                    workMinutes = 25;
                    isWorking = false;
                    modeLabel.Text = "休憩時間";
                    System.Media.SystemSounds.Beep.Play();
                }
            }
            else
            {
                breakMinutes--;
                if (breakMinutes < 0)
                {
                    breakMinutes = 5;
                    isWorking = true;
                    modeLabel.Text = "作業時間";
                    System.Media.SystemSounds.Beep.Play();
                }
            }

            timerLabel.Text = $"{Math.Max(workMinutes, breakMinutes):D2}:00";
        }

        private void StartButton_Click(object sender, EventArgs e)
        {
            isRunning = !isRunning;
            startButton.Text = isRunning ? "停止" : "開始";
            if (isRunning)
            {
                timer.Start();
            }
            else
            {
                timer.Stop();
            }
        }

        private void ResetButton_Click(object sender, EventArgs e)
        {
            isRunning = false;
            startButton.Text = "開始";
            timer.Stop();
            workMinutes = 25;
            breakMinutes = 5;
            isWorking = true;
            modeLabel.Text = "作業時間";
            timerLabel.Text = "25:00";
        }

        [STAThread]
        static void Main()
        {
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
            Application.Run(new PomodoroTimerForm());
        }
    }
}

File: pomodoro_timer.py
import tkinter as tk
from tkinter import ttk
import time
from datetime import timedelta
import threading

class PomodoroTimer:
    def __init__(self, root):
        self.root = root
        self.root.title("ポモドーロタイマー")
        self.root.geometry("300x200")
        
        # タイマーの初期設定
        self.work_minutes = 25
        self.break_minutes = 5
        self.is_working = True
        self.is_running = False
        
        # タイマーラベル
        self.timer_label = ttk.Label(root, text="25:00", font=("Helvetica", 48))
        self.timer_label.pack(pady=20)
        
        # スタート/ストップボタン
        self.start_button = ttk.Button(root, text="開始", command=self.toggle_timer)
        self.start_button.pack(pady=5)
        
        # リセットボタン
        self.reset_button = ttk.Button(root, text="リセット", command=self.reset_timer)
        self.reset_button.pack(pady=5)
        
        # モードラベル
        self.mode_label = ttk.Label(root, text="作業時間", font=("Helvetica", 12))
        self.mode_label.pack(pady=5)
        
    def update_timer(self):
        if self.is_running:
            if self.is_working:
                minutes = self.work_minutes
            else:
                minutes = self.break_minutes
            
            current_time = timedelta(minutes=minutes)
            self.timer_label.config(text=str(current_time))
            
            if minutes > 0:
                minutes -= 1
                if self.is_working:
                    self.work_minutes = minutes
                else:
                    self.break_minutes = minutes
                
                self.root.after(60000, self.update_timer)  # 1分ごとに更新
            else:
                # モードを切り替え
                self.is_working = not self.is_working
                if self.is_working:
                    self.work_minutes = 25
                    self.mode_label.config(text="作業時間")
                else:
                    self.break_minutes = 5
                    self.mode_label.config(text="休憩時間")
                
                # アラーム音を鳴らす
                self.beep()
                self.update_timer()
    
    def toggle_timer(self):
        self.is_running = not self.is_running
        if self.is_running:
            self.start_button.config(text="停止")
            self.update_timer()
        else:
            self.start_button.config(text="開始")
    
    def reset_timer(self):
        self.is_running = False
        self.start_button.config(text="開始")
        self.work_minutes = 25
        self.break_minutes = 5
        self.is_working = True
        self.mode_label.config(text="作業時間")
        self.timer_label.config(text="25:00")
    
    def beep(self):
        # システムのビープ音を鳴らす
        print("\a")

if __name__ == "__main__":
    root = tk.Tk()
    app = PomodoroTimer(root)
    root.mainloop()
